compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/TypesFactory.java

```diff
@@ -107,6 +112,16 @@ public static BallerinaTypeDescriptor getTypeDescriptor(BType bType, boolean raw
                 return new BallerinaTypeDescTypeDescriptor(moduleID, (BTypedescType) bType);
             case NIL:
                 return new BallerinaNilTypeDescriptor(moduleID, (BNilType) bType);
+            case FINITE:
```

Comment:
Don't we need to update the `getTypeDescKind` method?

Reply:
For the singleton type? The kind() method is implemented in the abstract type desc class. And the type desc kind gets passed to it when creating an instance of the typedesc.